alchemist/alchemist-implementationbase/src/main/java/it/unibo/alchemist/model/implementations/timedistributions/Trigger.java

```diff
@@ -41,9 +42,15 @@ protected void updateStatus(final Time curTime, final boolean executed, final do
         if (dryRunDone && curTime.compareTo(getNextOccurence()) >= 0 && executed) {
             setTau(new DoubleTime(Double.POSITIVE_INFINITY));
         }
+        this.executed = executed;
         dryRunDone = true;
     }
 
+    @Override
+    public boolean canOccurAgain() {
+        return !(executed && this.getNextOccurence().isInfinite());
```

Comment:
Uhmm... this smells. Why don't we just make sure that it we ran this when expected in updateStatus?